content/content/oop_macro.md

```diff
@@ -154,33 +154,37 @@ macro class*(head: expr, body: stmt): stmt {.immediate.} =
   #             Empty
 
   result.insert(0,
-    if baseName == nil:
-      quote do:
-        type `typeName` = ref object of RootObj
-    else:
-      quote do:
-        type `typeName` = ref object of `baseName`
+    newNimNode(nnkTypeSection).add(
```

Comment:
I don't like manually constructing the AST like this. Its harder to read and is much less concise.

I'd rather you improve the old `quote …`.

Maybe pull `baseName` into something like `let baseNameIdent = if baseName == nil: newIdent… else: …` instead? And then just have one `quote: …`